astropy/io/fits/convenience.py

```diff
@@ -433,15 +434,24 @@ def writeto(filename, data, header=None, output_verify='exception',
                 checksum=checksum)
 
 
-def table_to_hdu(table):
+def table_to_hdu(table, astropy_native=False):
```

Comment:
Actually it is more logical to provide the option of not writing full precision when the user does not explicitly want so. So ``astropy_native=True`` by default seems better. 

Just I am not able to comprehend whether users would find this easy to understand.
Nevertheless I am FOR the idea of having ``astropy_native=True``!!